src/OpenTelemetry/Metrics/ExponentialHistogramBuckets.cs

```diff
@@ -14,17 +14,21 @@
 // limitations under the License.
 // </copyright>
 
+#nullable enable
+
 namespace OpenTelemetry.Metrics;
 
-internal class ExponentialBuckets
+public class ExponentialHistogramBuckets
```

Comment:
It's probably better to start with a sealed public class first.

Reply:
I was actually just pondering this... I think I might make both `ExponentialBucketData` and `ExponentialHistogramBuckets` structs.

Reply:
Changed to structs in https://github.com/open-telemetry/opentelemetry-dotnet/pull/4290/commits/0cd360069d8eb4d1ba7756e0fc7f2cd6b8a7feac. As a result an implicit parameterless constructor was added to the public API... I'm wondering if I should explicitly make the parameterless constructor internal?

Reply:
Does it allow you to make the parameterless ctor for a struct internal?

Reply:
> Does it allow you to make the parameterless ctor for a struct internal?

Pretty sure parameterless ctor must be public 😢

> We did not do this for the HistogramBucket struct... maybe we should have?

I don't think it is needed there because we don't "new" it each time someone asks for the data, we just return the same instance over and over.

Reply:
Ok so after all that... I changed them back to classes 😆 .. sorry

It keeps the API simple and also instead of newing things up I reuse the same instance of `ExponentialHistogramData` on each snapshot.

I chatted with @CodeBlanch a bit offline about this approach. _Maybe_ there's some concern if an exporter were to hold onto one of these snapshots? I'm not sure though, it seems that if this is an issue, we have it elsewhere as well. I think we just need to be explicit that exporters _should not_ do such things and if for some reason they need to then it's their responsibility to make a copy if necessary.

That said, we can continue mulling this over... if we decide that the struct approach is better, we can change it again before releasing exp histograms stable.